### ISSUE TITLE
Embedded cache DB alternative

### ISSUE THREAD

**Issue by pditommaso** [nextflow-io/nextflow#2774]
## Bug report 

Nextflow tasks metadata is stored into a local embedded key-value database based on [LevelDB](https://github.com/dain/leveldb). 

This provides good performance, however, the LevelDB store has some stability issues on specific hardware/file systems causing and represent blocking factor for those users. See for example: https://github.com/nextflow-io/nextflow/issues/2377, https://github.com/nextflow-io/nextflow/issues/403, https://github.com/nextflow-io/nextflow/issues/351 and https://github.com/nextflow-io/nextflow/issues/309

The goal of this issue is to explore the use of [lmdbjava](https://github.com/lmdbjava/lmdbjava) as alternative storage for nextflow tasks metadata

**Comment by bentsherman** [comment]
I guess this one is resolved by the cloud cache #4097 

I know we were also considering parquet, but parquet is a columnar storage format so wouldn't be a good fit for the task cache. Instead any cache backend should be a true key-value store or at least row-based.

**Comment by bentsherman** [comment]
Although I see you were looking into LMDB. That should be a good choice. I can look into it if you want

**Comment by pditommaso** [comment]
I gave a try in the past to Lmdb in the past I was not really convinced: weird API, native OS dependencies, also it does not even really really [maintained any more](https://github.com/lmdbjava/lmdbjava/graphs/code-frequency). 

Maybe we should give a try to thin wrapper over classic Sqlite or [Duckbd](https://duckdb.org/)

**Comment by bentsherman** [comment]
That's too bad, LMDB seems to have really good performance. SQLite could be a good option. Probably not DuckDB though, since it is designed for OLAP. It could be useful to export a cache DB to DuckDB for downstream analytics, but not during the pipeline execution.

**Comment by pditommaso** [comment]
Only a benchmark can really tell

**Comment by pditommaso** [comment]
This should be prioritised. Java 24 is reporting these warning 

```
WARNING: A terminally deprecated method in sun.misc.Unsafe has been called
WARNING: sun.misc.Unsafe::invokeCleaner has been called by org.iq80.leveldb.util.ByteBufferSupport (file:/Users/pditommaso/.nextflow/framework/25.03.1-edge/nextflow-25.03.1-edge-one.jar)
WARNING: Please consider reporting this to the maintainers of class org.iq80.leveldb.util.ByteBufferSupport
WARNING: sun.misc.Unsafe::invokeCleaner will be removed in a future release
```

**Comment by jagedn** [comment]
yes, I've been playing with a "pure" postgre implementation (see https://edn-es.github.io/nf-pgcache/ ) and tested not only with a local postgre but also against Supadatabase or Neon, two cloud providers. This plugin is included in the plugin repository so you can give a try

also I was playing with a duckdb implementation and it capability to manage not only duckdb format but also sqlite for example (see https://incsteps.github.io/nf-duckdbcache/ ) 

related with them I've deployed also nf-cachebrowser, a web server plugin to navigate across cache. Right now only LevelDB implementation is tested but I would like to include postgre and duckdb

**Comment by pditommaso** [comment]
postgres is not exactly what I was suggesting into this issue 😸

**Comment by jagedn** [comment]
I used sqlite with duckdb extension (see https://incsteps.github.io/nf-duckdbcache/#_sqlite) 

after played a little with these different implementations, I think will not be very difficult to implement a "native" sqlite or duckdb version

**Comment by pditommaso** [comment]
I've gave a try to this https://github.com/nextflow-io/nextflow/pull/6255. We should run a benchmark to assess the impact of SQLite vs LevelDB.
